api_tests/applications/views/test_application_detail.py

```diff
@@ -39,19 +53,25 @@ def setUp(self):
                 }
             }
         }
+        return missing_id
 
-        self.missing_type = {
+    @pytest.fixture()
+    def missing_type(self, user_one_app):
```

Comment:
There's a lot of duplication in the payload fixtures, and the names aren't always obvious. A useful pattern would be to create a "function fixture", i.e. a fixture that returns a function. Something like:

```python
class TestApplication...:

    @pytest.fixture()
    def make_payload(self, user_app):
        """Function that returns a valid application payload by default
        Can pass type and ID to modify the payload.
        """
        def func(type='applications', id=user_app.client_id):
            return {
                'data': {
                    'id': id,
                    'type': type,
                    'attributes': {
                        #...
                    }
                }
            }
        return func

    def test_update_application(self, make_payload, ...):
        # make_payload returns a valid payload by default
        valid = make_payload()
        # ...

        invalid_type = make_payload(type='invalid')
        # ...

        missing_type = make_payload()
        del missing_type['type']
        # ...
```